validator.go

```diff
@@ -92,7 +91,7 @@ func newValidatorImpl(domains []string, usersFile string,
 		}
 		email = strings.ToLower(email)
 		for _, domain := range domains {
-			valid = valid || strings.HasSuffix(email, domain)
+			valid = valid || (strings.HasPrefix(domain, ".") && strings.HasSuffix(email, domain)) || strings.HasSuffix(email, "@"+domain)
```

Comment:
This might be best as a helper method to contain all the complexity in this check.

In the first case (wildcard subdomain denoted by a `.` prefix), I feel like we still need to make sure there's an `@`. Not sure how I would abuse it missing as a hacker, but it _feels_ like I can 😅

Reply:
(In what will be the new helper) Maybe use `strings.SplitN(email, "@", 2)` and check the second element with the suffix check?

Not sure if we want explicit `splitN...2` which assumes only a single `@` or if we should just do an uncapped split and just take the last element in the split slice for the suffix check.

Reply:
Created a helper method that contains the logic of validating the email agains the emailDomain. 

I choose to use the uncapped split by `@` and to check the last item in order to support backward compatibility (some teams might have used an user that doesn't contain just one `@`).